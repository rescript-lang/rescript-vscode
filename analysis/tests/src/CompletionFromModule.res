module SomeModule = {
  type t = {name: string}

  @get external getName: t => string = "name"

  let thisShouldNotBeCompletedFor = () => "hi"
}

let n = {SomeModule.name: "hello"}

// ^dv+
// n.
//   ^com
// ^dv-

@mainTypeForModule(SomeOtherModule)
type typeOutsideModule = {nname: string}

module SomeOtherModule = {
  type t = typeOutsideModule

  type irrelevantType = string

  @get external getNName: t => string = "nname"
  @get external getNName2: typeOutsideModule => string = "nname"
  @get external getNName3: irrelevantType => string = "nname"

  let thisShouldNotBeCompletedFor = () => "hi"
}

let nn: SomeOtherModule.t = {nname: "hello"}

// ^dv+
// nn.
//    ^com
// ^dv-

// @mainTypeForModule(SomeOthe) type typeOutsideModule = {nname: string}
//                            ^com
