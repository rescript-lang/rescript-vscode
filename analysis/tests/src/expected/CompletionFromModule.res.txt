
Complete src/CompletionFromModule.res 11:5
posCursor:[11:5] posNoWhite:[11:4] Found expr:[11:3->11:5]
Pexp_field [11:3->11:4] _:[15:0->11:5]
[set_result] set new result to Cpath Value[n].""
Completable: Cpath Value[n].""
Package opens Pervasives.JsxModules.place holder
Resolved opens 1 pervasives
ContextPath Value[n].""
[ctx_path]--> CPField
ContextPath Value[n]
[ctx_path]--> CPId
Path n
Path SomeModule.
[{
    "label": "->SomeModule.getName",
    "kind": 12,
    "tags": [],
    "detail": "t => string",
    "documentation": null,
    "sortText": "getName",
    "textEdit": {
    "range": {"start": {"line": 11, "character": 4}, "end": {"line": 11, "character": 4}},
    "newText": "->SomeModule.getName"
  }
  }, {
    "label": "name",
    "kind": 5,
    "tags": [],
    "detail": "string",
    "documentation": {"kind": "markdown", "value": "```rescript\nname: string\n```\n\n```rescript\ntype t = {name: string}\n```"}
  }]



Complete src/CompletionFromModule.res 33:6
posCursor:[33:6] posNoWhite:[33:5] Found expr:[33:3->33:6]
Pexp_field [33:3->33:5] _:[40:0->33:6]
[set_result] set new result to Cpath Value[nn].""
Completable: Cpath Value[nn].""
Package opens Pervasives.JsxModules.place holder
Resolved opens 1 pervasives
ContextPath Value[nn].""
[ctx_path]--> CPField
ContextPath Value[nn]
[ctx_path]--> CPId
Path nn
Path SomeOtherModule.
[{
    "label": "->SomeOtherModule.getNName",
    "kind": 12,
    "tags": [],
    "detail": "t => string",
    "documentation": null,
    "sortText": "getNName",
    "textEdit": {
    "range": {"start": {"line": 33, "character": 5}, "end": {"line": 33, "character": 5}},
    "newText": "->SomeOtherModule.getNName"
  }
  }, {
    "label": "->SomeOtherModule.getNName2",
    "kind": 12,
    "tags": [],
    "detail": "typeOutsideModule => string",
    "documentation": null,
    "sortText": "getNName2",
    "textEdit": {
    "range": {"start": {"line": 33, "character": 5}, "end": {"line": 33, "character": 5}},
    "newText": "->SomeOtherModule.getNName2"
  }
  }, {
    "label": "nname",
    "kind": 5,
    "tags": [],
    "detail": "string",
    "documentation": {"kind": "markdown", "value": "```rescript\nnname: string\n```\n\n```rescript\ntype typeOutsideModule = {nname: string}\n```"}
  }]


Complete src/CompletionFromModule.res 37:30
XXX Not found!
Completable: Cpath Module[SomeOthe]
Package opens Pervasives.JsxModules.place holder
Resolved opens 1 pervasives
ContextPath Module[SomeOthe]
Path SomeOthe
[{
    "label": "SomeOtherModule",
    "kind": 9,
    "tags": [],
    "detail": "module SomeOtherModule",
    "documentation": null
  }]

